main.tf

```diff
@@ -52,124 +78,62 @@ resource "aws_wafv2_web_acl" "main" {
+                dynamic "all_query_arguments" {
+                  for_each = length(lookup(field_to_match.value, "all_query_arguments", {})) == 0 ? [] : [lookup(field_to_match.value, "all_query_arguments")]
+                  content {}
+                }
+                dynamic "body" {
+                  for_each = length(lookup(field_to_match.value, "body", {})) == 0 ? [] : [lookup(field_to_match.value, "body")]
+                  content {}
+                }
+                dynamic "method" {
+                  for_each = length(lookup(field_to_match.value, "method", {})) == 0 ? [] : [lookup(field_to_match.value, "method")]
+                  content {}
+                }
+                dynamic "query_string" {
+                  for_each = length(lookup(field_to_match.value, "query_string", {})) == 0 ? [] : [lookup(field_to_match.value, "query_string")]
+                  content {}
+                }
+              }
+            }
+            positional_constraint = lookup(byte_match_statement.value, "positional_constraint")
+            search_string         = lookup(byte_match_statement.value, "search_string")
+            text_transformation {
+              priority = lookup(byte_match_statement.value, "priority")
+              type     = lookup(byte_match_statement.value, "type")
+            }
+          }
         }
-      }
 
-      statement {
+        # geo_match_statement
         dynamic "geo_match_statement" {
           for_each = length(lookup(rule.value, "geo_match_statement", {})) == 0 ? [] : [lookup(rule.value, "geo_match_statement", {})]
           content {
             country_codes = lookup(geo_match_statement.value, "country_codes")
           }
         }
-      }
-
-      dynamic "visibility_config" {
-        for_each = length(lookup(rule.value, "visibility_config")) == 0 ? [] : [lookup(rule.value, "visibility_config", {})]
-        content {
-          cloudwatch_metrics_enabled = lookup(visibility_config.value, "cloudwatch_metrics_enabled", true)
-          metric_name                = lookup(visibility_config.value, "metric_name", "${var.name_prefix}-geo-match-metric-name")
-          sampled_requests_enabled   = lookup(visibility_config.value, "sampled_requests_enabled", true)
-        }
-      }
-    }
-  }
-
-  dynamic "rule" {
-    for_each = var.ip_set_rules
-    content {
-      name     = lookup(rule.value, "name")
-      priority = lookup(rule.value, "priority")
-
-      action {
-        dynamic "allow" {
-          for_each = length(lookup(rule.value, "action", {})) == 0 || lookup(rule.value, "action", {}) == "allow" ? [1] : []
-          content {}
-        }
-
-        dynamic "count" {
-          for_each = lookup(rule.value, "action", {}) == "count" ? [1] : []
-          content {}
-        }
-
-        dynamic "block" {
-          for_each = lookup(rule.value, "action", {}) == "block" ? [1] : []
-          content {}
-        }
-      }
 
-      statement {
+        # ip_set_statement
```

Comment:
I don't think we need this comment here, it is clear from dynamic block below what it is

Reply:
Cleaned up commenting a bit